cpp/ql/src/Likely Bugs/OO/UnsafeUseOfThis.qhelp

```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+
+<overview>
+<p>This rule finds calls to pure virtual member functions in constructors and destructors. Such calls do not perform virtual dispatch, and can cause undefined behavior.</p>
+
+</overview>
+<recommendation>
+<p>Do not rely on virtual dispatch in constructors and destructors. Instead, each class should be responsible for acquiring and releasing its resources.</p>
+
+</recommendation>
+<example><sample src="UnsafeUseOfThis.cpp" />
+
+
+
+</example>
+<references>
+
+<li>
+  <a href="https://isocpp.org/wiki/faq/strange-inheritance#calling-virtuals-from-ctors">When my base class’s constructor calls a virtual function on its this object, why doesn’t my derived class’s override of that virtual function get invoked?</a>
```

Comment:
Please clarify the source of this reference: ISO C++ FAQ.

Reply:
The other queries we have refer to isocpp as `C++ FAQ`. Is that more appropriate, or should we update them (in a separate PR) to be `ISO C++ FAQ`?

Reply:
Unification sounds good. I have no preference for which name it should be, as long as it sounds official. Apparently the FAQ is really called [the C++ Super-FAQ](https://isocpp.org/wiki/faq), but that doesn't sound official or authoritative!

Reply:
I went with `ISO C++ FAQ` in 4a7f9100e41d58cb7f8a96784307ce40fd2a4aee. I'll update the other ones later today.